smt/applications/mfk.py

```diff
@@ -958,25 +958,27 @@ def _check_param(self):
                 )
 
         n_param = d
-        if self.options["categorical_kernel"] in [
-            MixIntKernelType.EXP_HOMO_HSPHERE,
-            MixIntKernelType.HOMO_HSPHERE,
-            MixIntKernelType.CONT_RELAX,
-        ]:
-            n_comp = self.options["n_comp"] if "n_comp" in self.options else None
-            mat_dim = (
-                self.options["cat_kernel_comps"]
-                if "cat_kernel_comps" in self.options
-                else None
-            )
 
-            n_param = compute_n_param(
-                self.design_space,
-                self.options["categorical_kernel"],
-                d,
-                n_comp,
-                mat_dim,
-            )
+        if self.options["categorical_kernel"] is not None:
```

Comment:
This test can be removed

Reply:
If we use MFK with continuous variables only, we will not set self.options["categorical_kernel"]. Then we will have self.options["categorical_kernel"] = None.  In this case removing the test "if self.options["categorical_kernel"] is not None:" will lead to raise the error "Only the continuous relaxation kernel is available with multi-fidelity" whereas we don't need to set a categorical kernel since only continuous variables are considered.

Reply:
Oups miss the `else`! Ok!